programming/python/libraries.md

```diff
@@ -11,69 +11,41 @@ Robot Kit
- * [mpmath 0.18](https://code.google.com/p/mpmath/)
- * [nose 1.3.0](http://nose.readthedocs.org/)
- * [nss 0.15.0](https://pypi.python.org/pypi/python-nss)
- * [ntplib 0.3.1](https://pypi.python.org/pypi/ntplib/)
- * [numexpr 2.2.2](https://github.com/pydata/numexpr)
- * [NumPy 1.8.2](http://www.numpy.org/)
- * [pandas 0.10.0](http://pandas.pydata.org/)
- * [pexpect 3.1](http://pexpect.readthedocs.org/en/latest/)
- * [pillow 2.2.1](http://pillow.readthedocs.org/)[^1]
- * [pwquality 1.2.3](https://apps.fedoraproject.org/packages/python-pwquality)
- * [pycurl 7.19.3](http://pycurl.sourceforge.net/)
- * [Pygments 1.6](http://pygments.org/)
- * [pygobject2 2.28.6](https://apps.fedoraproject.org/packages/pygobject2)
- * [pygpgme 0.3](https://pypi.python.org/pypi/pygpgme/0.3)
- * [pykickstart 1.99.48](https://fedoraproject.org/wiki/Pykickstart)[^2]
- * [pyliblzma 0.5.3](https://pypi.python.org/pypi/pyliblzma)
- * [Pyparsing 2.0.1](http://pyparsing.wikispaces.com/)
- * [PyParted 3.9](https://fedorahosted.org/pyparted/)
- * [pySerial 2.6](http://pyserial.sourceforge.net/)
- * [pytalloc 2.1.0](https://apps.fedoraproject.org/packages/pytalloc)
- * [pytz 2012d-5](http://pytz.sourceforge.net/)
- * [pyudev 0.15](https://pyudev.readthedocs.org)
- * [PyUSB 1.0.0](http://pyusb.sourceforge.net/)
- * [pyxattr 0.5.1](http://pyxattr.k1024.org/)
- * [PyXDG 0.25](http://freedesktop.org/wiki/Software/pyxdg)
- * [PyYAML 3.10](http://pyyaml.org/)
- * [SciPy 0.12.1](http://www.scipy.org/)
- * [simplegeneric 0.8](https://pypi.python.org/pypi/simplegeneric)
- * [simplejson 3.5.3](https://pypi.python.org/pypi/simplejson)
- * [six 1.7.3](https://pypi.python.org/pypi/six)
- * [slip 0.6.0](https://fedorahosted.org/python-slip/)
- * [slip-dbus 0.6.0](https://apps.fedoraproject.org/packages/python-slip-dbus)
- * [sssdconfig 1.11.6](https://apps.fedoraproject.org/packages/python-sssdconfig)
- * [SymPy 0.7.5](http://sympy.org/)
- * [tables 3.0.0](https://pypi.python.org/pypi/tables)
- * [Tornado 2.2.1](http://www.tornadoweb.org/)
- * [urlgrabber 3.10.1](http://urlgrabber.baseurl.org)
- * [Werkzeug 0.9.4](http://werkzeug.pocoo.org/)
- * [ZeroMQ 13.0.2](http://zeromq.org/bindings:python)
+* [astoria 0.7.0](https://pypi.org/project/astoria)
+* [cached-property 1.5.2](https://pypi.org/project/cached-property)
+* [click 8.0.1](https://pypi.org/project/click)
+* [cycler 0.10.0](https://pypi.org/project/cycler)
+* dateutil 2.8.2
+* dateutil-zoneinfo 2.8.2
+* [dbus-next 0.2.3](https://pypi.org/project/dbus-next)
+* [docopt 0.6.2](https://pypi.org/project/docopt)
+* [gmqtt 0.6.10](https://pypi.org/project/gmqtt)
+* [gpg 1.16.0](https://pypi.org/project/gpg)
+* [iniparse 0.5](https://pypi.org/project/iniparse)
+* [j5 0.13.0](https://pypi.org/project/j5)
+* [j5-zoloto 0.1.0](https://pypi.org/project/j5-zoloto)
+* [kiwisolver 1.3.2](https://pypi.org/project/kiwisolver)
+* [matplotlib 3.4.1](https://pypi.org/project/matplotlib)
+* [numpy 1.21.2](https://pypi.org/project/numpy)
+* opencv 4.5.2
```

Comment:
```suggestion
* [opencv 4.5.2](https://pypi.org/project/opencv-python-headless/)
```
Presumably?

Reply:
We're actually compiling OpenCV ourselves, so are not using the pre-built binary you've suggested. Not actually sure what to do with this one.

Reply:
Ah, is there a Python API available? If not, maybe just link out to OpenCV's main homepage?

Reply:
So the python bindings for OpenCV are generated using SWIG, so there isn't actually source to link to there.

I'm keen to keep this automated if possible, so might add a section with some hard-coded sites to that script.